src/main/java/com/amazon/opendistroforelasticsearch/knn/index/KNNVectorFieldMapper.java

```diff
@@ -242,6 +242,10 @@ public void parse(ParseContext context) throws IOException {
                                                     "update knn.plugin.enabled setting to true");
         }
 
+        if (KNNSettings.isCircuitBreakerTriggered()) {
+            throw new IllegalStateException("Cannot update KNN vector field when circuit breaker is triggered.");
```

Comment:
can we change message on the following lines "Indexing knn vector fields is rejected as circuit breaker triggered. Check _opendistro/_knn/stats for detailed state"

Reply:
Sure.